models/applications.ts

```diff
@@ -1,6 +1,56 @@
 import { application } from "../types/application";
 const firestore = require("../utils/firestore");
 const ApplicationsModel = firestore.collection("applicants");
+const { DOCUMENT_WRITE_SIZE: FIRESTORE_BATCH_OPERATIONS_LIMIT } = require("../constants/constants");
+import { chunks } from "../utils/array";
+
+const batchUpdateApplications = async () => {
+  try {
+    const operationStats = {
+      failedApplicationUpdateIds: [],
+      totalFailedApplicationUpdates: 0,
+      totalApplicationUpdates: 0,
+    };
+
+    const updatedApplications = [];
+    const applications = await ApplicationsModel.get();
+
+    if (applications.empty) {
+      return operationStats;
+    }
+
+    operationStats.totalApplicationUpdates = applications.size;
+
+    applications.forEach((application) => {
+      const taskData = application.data();
+      taskData.createdAt = null;
+      updatedApplications.push({ id: application.id, data: taskData });
+    });
+
+    const multipleApplicationUpdateBatch = [];
+    const chunkedApplication = chunks(updatedApplications, FIRESTORE_BATCH_OPERATIONS_LIMIT);
+
+    chunkedApplication.forEach(async (applications) => {
+      const batch = firestore.batch();
+      applications.forEach(({ id, data }) => {
+        batch.update(ApplicationsModel.doc(id), data);
+      });
+      try {
+        await batch.commit();
```

Comment:
Just asking, do we have the limit on the number of record being saved at a time?